tests/ihontaryk/homework_6/test_unpack.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+from ihontaryk.homework_6.unpack import unpack_while_loop, unpack_recursive, unpack_while_loop_gen, unpack_recursive_gen
+
+
+@pytest.mark.parametrize('arguments, expected_result',
+                         [([123, ["234", None], {"sdf": True, "bb": 44}, {(1, 2, 3): "abc"}],
+                           [123, '234', None, 'sdf', True, 'bb', 44, 1, 2, 3, 'abc'])
+                          ])
+def test_unpack_while_loop(arguments, expected_result):
+    """
+    verify unpack_while_loop function
+    """
+
+    assert unpack_while_loop(arguments) == expected_result
+
+
+@pytest.mark.parametrize('arguments, expected_result',
+                         [([123, ["234", None], {"sdf": True, "bb": 44}, {(1, 2, 3): "abc"}],
+                           [123, '234', None, 'sdf', True, 'bb', 44, 1, 2, 3, 'abc'])
+                          ])
+def test_unpack_recursive(arguments, expected_result):
+    """
+    verify unpack_recursive function
+    """
+
+    assert unpack_recursive(arguments) == expected_result
+
+
+@pytest.mark.parametrize('arguments, expected_result',
+                         [([123, ["234", None], {"sdf": True, "bb": 44}, {(1, 2, 3): "abc"}],
+                           [123, '234', None, 'sdf', True, 'bb', 44, 1, 2, 3, 'abc'])
+                          ])
+def test_unpack_while_loop_gen(arguments, expected_result):
+    """
+    verify unpack_while_loop_gen function
+    """
+
+    assert list(unpack_while_loop_gen(arguments)) == expected_result
+
+
+@pytest.mark.parametrize('arguments, expected_result',
+                         [([123, ["234", None], {"sdf": True, "bb": 44}, {(1, 2, 3): "abc"}],
+                           [123, '234', None, 'sdf', True, 'bb', 44, 1, 2, 3, 'abc'])
+                          ])
```

Comment:
you try to make 1 test to match all cases 
since it is 1 case no need to use parametrize, 
and I would suggest to make couple small tests, which check some aspect of a function:

- only dict
- only list
- only simple items

etc,
so that idea of each case is clear, 
and if you make change in function, you know what aspect of function failed, 
with this test you will receive single failure and will have to dig into error log